File: packages/shared/src/commonMain/kotlin/com/opencount/shared/model/SessionTag.kt
package com.opencount.shared.model

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
data class SessionTag(
    val id: String,
    val name: String,
    val colorHex: String = "#3498DB",
    val emoji: String = "\uD83C\uDFF7\uFE0F",
    val createdAt: Instant,
) {
    companion object {
        val predefinedTags: List<Triple<String, String, String>> = listOf(
            Triple("Wildlife", "#2ECC71", "\uD83E\uDD81"),
            Triple("Inventory", "#3498DB", "\uD83D\uDCE6"),
            Triple("Research", "#9B59B6", "\uD83D\uDD2C"),
            Triple("Agriculture", "#F39C12", "\uD83C\uDF3E"),
            Triple("Urban", "#E74C3C", "\uD83C\uDFD9\uFE0F"),
            Triple("Marine", "#1ABC9C", "\uD83D\uDC20"),
            Triple("Aerial", "#34495E", "\u2708\uFE0F"),
            Triple("Medical", "#E67E22", "\uD83C\uDFE5"),
            Triple("Sports", "#F1C40F", "\u26BD"),
            Triple("Education", "#16A085", "\uD83D\uDCDA"),
        )
    }
}

File: apps/ios/app/Models/SessionTag.swift
import Foundation

// MARK: - SessionTag

final class SessionTag: ObservableObject, Identifiable, Codable {
    var id: UUID
    var name: String
    var colorHex: String
    var emoji: String
    var createdAt: Date

    init(id: UUID = UUID(), name: String, colorHex: String = "#3498DB",
         emoji: String = "🏷️", createdAt: Date = Date()) {
        self.id = id; self.name = name; self.colorHex = colorHex
        self.emoji = emoji; self.createdAt = createdAt
    }

    static let predefinedTags: [(name: String, colorHex: String, emoji: String)] = [
        ("Wildlife","#2ECC71","🦁"),("Inventory","#3498DB","📦"),
        ("Research","#9B59B6","🔬"),("Agriculture","#F39C12","🌾"),
        ("Urban","#E74C3C","🏙️"),("Marine","#1ABC9C","🐠"),
        ("Aerial","#34495E","✈️"),("Medical","#E67E22","🏥"),
        ("Sports","#F1C40F","⚽"),("Education","#16A085","📚"),
    ]
}
